query/src/predicate.rs

```diff
@@ -294,13 +307,77 @@ impl PredicateBuilder {
     pub fn build(self) -> Predicate {
         self.inner
     }
+
+    /// Returns all the possible push-downed expressions of the given filters in the
+    /// form of a Predicate
+    /// Currently, we only push down primitive binary expression in
+    /// this form: `column op constant`
+    pub fn pushdown_predicates(
+        self,
+        filters: &[Expr],
+    ) -> std::result::Result<Predicate, DataFusionError> {
+        // For each expression of the filters, recursively split it it if it is is an AND conjunction
+        // For example, expression (x AND y) will be split into 2 expression [x, y]
+        let mut exprs = vec![];
+        filters
+            .iter()
+            .for_each(|expr| Self::split_members(expr, &mut exprs));
+
+        // Only keep single_column and primitive binary expressions
+        let mut pushdown_exprs: Vec<Expr> = vec![];
+        exprs
+            .into_iter()
+            .try_for_each::<_, std::result::Result<_, DataFusionError>>(|expr| {
+                let mut columns: HashSet<String> = HashSet::new();
```

Comment:
```suggestion
                let mut columns = HashSet::new();
```

You should be able to elide the type annotation here.